src/server/main_service.cc

```diff
@@ -1590,8 +1572,27 @@ void Service::Exec(CmdArgList args, ConnectionContext* cntx) {
     }
   }
 
-  bool scheduled =
-      StartMultiExec(cntx->db_index(), cntx->transaction, &exec_info, &tmp_keys, global_script);
+  bool transactional = global_script;
```

Comment:
we can probably factor out the logic into a separate helper function.

Reply:
We should, Exec is large and convoluted, this is why I kept it in StartMultiExec in the first place

Reply:
fixed.